### PR TITLE
feat: add creation user required step

### PR DESCRIPTION
**Pull request by yassinedorbozgithub**
image

## Summary by CodeRabbit

- **New Features**
	- Introduced the `RequiredSteps` component for user authentication and account creation.
	- Added the `AddUserDialog` component for user creation via a dialog interface.
- **Bug Fixes**
	- Improved user feedback and error handling during the user creation process.
- **Refactor**
	- Simplified layout structure by replacing the `Grid` component with the `RequiredSteps` component.
- **API Enhancements**
	- Expanded API capabilities to include user management operations.

### CODE REVIEW

File: src/app/components/RequiredSteps.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+export const RequiredSteps = ({ children }: { children: React.ReactNode }) => {
+  const { user, isAuthenticated } = useAuth();
+  const pathname = usePathname();
+
+  const [isCreateUserDialogOpened, setIsCreateUserDialogOpened] =
+    React.useState(false);
+  const [hasUser, setHasUser] = React.useState(false);
+  const isAnonymous = !isAuthenticated && pathname === '/viewer';
+
+  console.log(pathname);
```
Comment (jacob-khoza-symb) [inline src/app/components/RequiredSteps.tsx:19]: debug code?

**Review by jacob-khoza-symb — APPROVED** [review]

File: src/app/components/RequiredSteps.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import { AddUserDialog } from './dialogs/AddUserDialog';
+import { useAuth } from '../context/AuthProvider';
+import { apiUser } from '../services/endpoints/user.class';
+
+export const RequiredSteps = ({ children }: { children: React.ReactNode }) => {
+  const { user, isAuthenticated } = useAuth();
+  const pathname = usePathname();
+
+  const [isCreateUserDialogOpened, setIsCreateUserDialogOpened] =
+    React.useState(false);
```
Comment (amalessid) [inline src/app/components/RequiredSteps.tsx:15]: Hello @yassinedorbozgithub, I suggest we import everything we need from react  once and for all, to keep the code aligned with other files.

```suggestion
    useState(false);
```

File: src/app/components/RequiredSteps.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const [hasUser, setHasUser] = React.useState(false);
+  const isAnonymous = !isAuthenticated && pathname === '/viewer';
+
+  console.log(pathname);
+
+  React.useEffect(() => {
```
Comment (amalessid) [inline src/app/components/RequiredSteps.tsx:19]:
```suggestion
  useEffect(() => {
```

File: src/app/components/dialogs/AddUserDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
+'use client';
+
+import { KeyboardBackspace, Send } from '@mui/icons-material';
```
Comment (amalessid) [inline src/app/components/dialogs/AddUserDialog.tsx:3]:
```suggestion
import {  Send } from '@mui/icons-material';
```

File: src/app/components/dialogs/AddUserDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+import {
+  Dialog,
+  DialogContent,
+  Grid,
+  Alert,
+  TextField,
+  Button,
```
Comment (amalessid) [inline src/app/components/dialogs/AddUserDialog.tsx:10]:
```suggestion
  
```

File: src/app/components/RequiredSteps.tsx (hunk 1 above)
Comment (yassinedorbozgithub) [reply src/app/components/RequiredSteps.tsx]: You are right the mentioned code need to be removed, nice catch

File: src/app/components/RequiredSteps.tsx (hunk 2 above)
Comment (yassinedorbozgithub) [reply src/app/components/RequiredSteps.tsx:15]: Sure we will do that

**Comment by yassinedorbozgithub** [comment]
Hi @amalessid, all the comments are addressed :white_check_mark:

File: src/app/components/RequiredSteps.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+  useEffect(() => {
+    const fetchUser = async (id: string) =>
+      apiUser
+        .getUser(id)
+        .then(({ data }) => {
+          setHasUser(true);
+          return data;
+        })
+        .catch((e) => {
```
Comment (amalessid) [inline src/app/components/RequiredSteps.tsx:27]: Hello @yassinedorbozgithub, Or maybe you prefer to log the error instead?
```suggestion
        .catch(() => {
```

Comment (yassinedorbozgithub) [reply src/app/components/RequiredSteps.tsx:27]: Hi @amalessid, i think that we can remove e prop like you suggest

**Review by BMartinos — APPROVED** [review]

**Pull request merged**